ddl/index.go

```diff
@@ -321,11 +320,22 @@ func (d *ddl) convert2RollbackJob(t *meta.Meta, job *model.Job, tblInfo *model.T
 	indexInfo.State = model.StateDeleteOnly
 	originalState := indexInfo.State
 	job.SchemaState = model.StateDeleteOnly
-	_, err := updateTableInfo(t, job, tblInfo, originalState)
-	if err != nil {
-		return ver, errors.Trace(err)
+	_, err1 := updateTableInfo(t, job, tblInfo, originalState)
+	if err1 != nil {
+		return ver, errors.Trace(err1)
+	}
+
+	if kv.ErrKeyExists.Equal(err) {
+		return ver, kv.ErrKeyExists.Gen("Duplicate for key %s", indexInfo.Name.O)
 	}
-	return ver, kv.ErrKeyExists.Gen("Duplicate for key %s", indexInfo.Name.O)
+
+	// Clean up the channel of notifyCancelReorgJob. Make sure it can't affect other jobs.
```

Comment:
It is possible that the reorg just finished, returned without any error, then the channel may affect the next job.

Move the cleanup code right after timeout error check would be safer.